google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/BigQueryImplTest.java

```diff
@@ -1144,21 +1145,24 @@ public void testQueryRequestCompleted() throws InterruptedException {
                 PROJECT, JOB, Collections.<BigQueryRpc.Option, Object>emptyMap()))
         .andReturn(jobResponsePb);
     EasyMock.expect(
-        bigqueryRpcMock.getQueryResults(PROJECT, JOB, optionMap)).andReturn(responsePb);
+            bigqueryRpcMock.listTableData(
+                PROJECT, DATASET, TABLE, Collections.<BigQueryRpc.Option, Object>emptyMap()))
+        .andReturn(Tuple.<String, Iterable<TableRow>>of("", ImmutableList.of(TABLE_ROW)));
 
     EasyMock.replay(bigqueryRpcMock);
     bigquery = options.getService();
-    QueryResult result =
+    // TODO(pongad) revert back to QueryResult
+    Page<FieldValueList> result =
         bigquery
             .query(QUERY_JOB_CONFIGURATION_FOR_QUERY, queryJob)
             .getQueryResults(QueryResultsOption.pageSize(42L));
-    assertEquals(null, result.getSchema());
-    assertEquals(1L, result.getTotalRows());
+    // assertEquals(null, result.getSchema());
+    // assertEquals(1L, result.getTotalRows());
```

Comment:
`tabledata.list` returns `totalRows` as well. I'm slightly surprised there isn't a custom type already for row data that includes this property. I've seen feedback on the Python API that people do use this property.

Reply:
I agree. Noted in #2591.